### PR TITLE
bd-7 feat: add input component

### PR DESCRIPTION
**Pull request by SlavaMryasov**

### CODE REVIEW

File: src/components/input/Input.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          {label}
+        </label>
+      )}
+      <div
+        className={cn(
+          'relative inline-flex h-9 items-center rounded-sm border border-dark-100 text-light-900',
+          'hover:border-light-900',
+          'active:border-light-100',
+          'active:text-light-100',
+          'focus-within:border-accent-500',
```
Comment (sfgardev) [inline src/components/input/Input.tsx:44]: use focus on input (semantically right to use it in input)
https://bharathvaj-ganesan.medium.com/focus-vs-focus-within-vs-focus-visible-2db18593495b

File: src/components/input/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          {type === 'password' && currentType === 'text' && (
+            <Icon color='light-100' icon='eye-outline' />
+          )}
+          {type === 'search' && (
+            <Icon color='light-100' icon='search-outline' />
+          )}
+        </div>
+      </div>
+      {errorText && (
+        <span className='block text-sm text-danger-500'>{errorText}</span>
```
Comment (sfgardev) [inline src/components/input/Input.tsx:78]: use p instead

File: src/components/input/Password.stories.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+export default meta
+const commonOptions: CommonOptions = {
+  label: 'password',
+  placeholder: 'enter password',
+  type: 'password',
+}
+
+export const Default: Story<typeof meta> = {
```
Comment (sfgardev) [inline src/components/input/Password.stories.ts:17]: use StoryObj, remove redundant abstraction

**Review by sfgardev — CHANGES_REQUESTED** [review]

File: src/components/input/Input.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+export type InputProps = {
+  errorText?: string
+  icon?: string
+  label?: string
+} & ComponentProps<'input'>
+
+export const Input = ({
+  disabled,
+  errorText,
+  icon,
```
[the remark this answers is not in the document]
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:15]: done

File: src/components/input/Input.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  label,
+  placeholder,
+  type,
+}: InputProps) => {
+  const [currentType, setCurrentType] = useState(type)
+  const id = useId()
```
[the remark this answers is not in the document]
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:21]: done

File: src/components/input/Input.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+  const setVisibleHandler = () => {
+    if (type === 'password') {
+      setCurrentType(currentType === 'password' ? 'text' : 'password')
+    }
+  }
+
+  const positionIcon = type === 'search' ? 'flex-row-reverse justify-end' : ''
```
[the remark this answers is not in the document]
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:29]: done

File: src/components/input/Input.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+          `${errorText && 'border-danger-500'}`
+        )}
+      >
+        <input
+          className={cn(
+            'm-1 w-60 border-none bg-transparent text-light-900',
+            'active:text-light-100',
+            'focus:bg-transparent focus:outline-none',
+            'disabled:cursor-not-allowed',
+            // 'disabled:custom-block-cursor',
```
[the remark this answers is not in the document]
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:56]: done

File: src/components/input/Input.tsx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  return (
+    <>
+      {label && (
+        <label className='block text-sm text-light-900' htmlFor={id}>
+          {label}
+        </label>
+      )}
+      <div
+        className={cn(
+          'relative inline-flex h-9 items-center rounded-sm border border-dark-100 text-light-900',
```
[the remark this answers is not in the document]
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:40]: remove inline-flex and add w-280

File: src/components/input/Input.tsx [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+          'hover:border-light-900',
+          'active:border-light-100',
+          'active:text-light-100',
+          'focus-within:border-accent-500',
+          `${positionIcon}`,
+          `${disabled && 'text-dark-100'}`,
```
[the remark this answers is not in the document]
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:46]: done for desabled

File: src/components/input/Input.tsx [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+            'disabled:cursor-not-allowed',
+            // 'disabled:custom-block-cursor',
+            `${errorText && 'text-light-100'}`,
+            `${disabled && 'text-dark-100'}`
+          )}
+          disabled={disabled}
+          id={id}
+          placeholder={placeholder}
+          type={currentType}
+        ></input>
```
[the remark this answers is not in the document]
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:64]: done

File: src/components/input/Input.tsx (hunk 1 above)
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:44]: done

File: src/components/input/Input.tsx (hunk 2 above)
Comment (SlavaMryasov) [reply src/components/input/Input.tsx:78]: done

File: src/components/input/Password.stories.ts (hunk 3 above)
Comment (SlavaMryasov) [reply src/components/input/Password.stories.ts:17]: done

**Comment by SlavaMryasov** [comment]
please review the corrections

File: src/components/input/Input.tsx [hunk 11]
```diff
@@ -35,48 +39,44 @@ export const Input = ({
...
-          'focus-within:border-accent-500',
-          `${positionIcon}`,
-          `${disabled && 'text-dark-100'}`,
-          `${errorText && 'border-danger-500'}`
-        )}
-      >
+      <div className={cn('relative h-9 max-w-[280px]', `${iconAlignment}`)}>
         <input
+          autoComplete='off' // for disabled dropdown list
           className={cn(
```
Comment (sfgardev) [inline src/components/input/Input.tsx:45]: you can write clsx classes lile this {'bg-red: errorText'}

File: src/components/input/Input.tsx [hunk 12]
```diff
@@ -35,48 +39,44 @@ export const Input = ({
...
-        <div className='m-1' onClick={setVisibleHandler}>
+          value={props.value}
+        />
+        <div
+          className={`absolute right-1 top-[2px] m-1 ${disabled ? '-z-10' : ''}`}
+          onClick={setVisibleHandler}
+        >
           {type === 'password' && currentType === 'password' && (
-            <Icon color='light-100' icon='eye-off-outline' />
+            <Icon className={iconColor} icon='eye-off-outline' />
```
Comment (sfgardev) [inline src/components/input/Input.tsx:69]: input component should be independent from icon (if eye-off-outline be deleted,  we need to fix input)

**Review by sfgardev — CHANGES_REQUESTED** [review]

**Review by sfgardev — APPROVED** [review]

**Review by itBanda — APPROVED** [review]

**Pull request closed without merging**